src/NRules/NRules.Fluent/Dsl/RuleBuildContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿using NRules.Fluent.Expressions;
+using NRules.RuleModel.Builders;
+
+namespace NRules.Fluent.Dsl;
+
+internal readonly struct RuleBuildContext
+{
+    private RuleBuildContext(RuleBuilder builder)
+    {
+        Builder = builder;
+    }
+
+    public static RuleBuildContext CreateNew(RuleMetadata metadata)
+    {
+        var context = new RuleBuildContext(new RuleBuilder());
+        context.ApplyMetadata(metadata);
+        return context;
+    }
+
+    public static RuleBuildContext Empty = new();
+
+    public SymbolStack SymbolStack { get; } = new();
+
+    public RuleBuilder Builder { get; }
+
+    public bool IsEmpty => Builder is null;
+
+    private void ApplyMetadata(RuleMetadata metadata)
```

Comment:
@gritcsenko I think it was better when it was inside the factory, no? The idea is that RuleBuilderContext is just a dumb wrapper, w/o any logic in it.

Reply:
Moved, thanks for notice